### DISCUSSION TITLE
How to add custom button in dashboard with custom data shown ?

### DISCUSSION THREAD

**Discussion by sergeynilov** [filamentphp/filament#6402] [Help] [answered]
In laravel 9 / filamenphp app I need to make site data checking by clicking on custom button in dashboard
and running checks functionality(it takes some time so I do not want to run it when dashboarrd is opened) 
    and opening results in modal custom page.

I made similar action in EditSubject class of SubjectResource :
```
<?php

class EditSubject extends EditRecord
{
    protected static string $resource = SubjectResource::class;

    protected function getActions(): array
    {
        return [
            Actions\DeleteAction::make(),
            Action::make('Preview')
                ->action(function (array $data): void {
                })
                ->modalContent(view('filament.pages.actions.Subject-preview', ['SubjectId' => $this->record->id]))
                ->modalHeading('Subject preview')
                ->modalSubheading('That is how this Subject would look on frontend pages')
                ->modalButton('Return to editor')
        ];
    }

```

where Subject-preview that is laravel component, so I run script in it and show results in filament.pages.actions.Subject-preview
file. But I do not know how to add custom action to dashboard?
in config/filament.php I have widgets block:

```
    'widgets' => [
        'namespace' => 'App\\Filament\\Widgets',
        'path' => app_path('Filament/Widgets'),
        'register' => [
            Widgets\AccountWidget::class,

```

But how can I create widget  with custom action only to run my checking functionality only on button clicking ?

Thank you!

**Comment by danharrin** [comment]
Create a page called Dashboard. Extend the base Dashboard class from Filament. Replace the old dashboard with your dashboard in the config file. Then use `getActions()` like normal on the new dashboard page.

**Reply by sergeynilov** [reply]
I created a file with command :
`php artisan make:filament-page Dashboard  --type=custom
`
and modified file app/Filament/Pages/Dashboard.php :

```
<?php

namespace App\Filament\Pages;

// With automatic import I found this line - not shure is it correct
use Filament\Facades\Filament;
// use Filament\Pages\Page;


//class Dashboard extends Page
class Dashboard extends Filament // Replaced to Filament - did you mean this ?
{
    protected static ?string $navigationIcon = 'heroicon-o-document-text';

    protected static string $view = 'filament.pages.dashboard';

    protected function getActions(): array
    {
        \Log::info(varDump(-1, ' -1 getActions  app/Filament/Pages/Dashboard.php::'));
        return [
        ];
    }

}

```

Modified config/filament.php :
```
    'pages' => [
        'namespace' => 'App\\Filament\\Pages',
        'path' => app_path('Filament/Pages'),
        'register' => [
//            Pages\Dashboard::class,
            App\Filament\Pages\Dashboard::class,
        ],
    ],

```
and cleared cache.

But I got error on “/admin” url:
`404  NOT FOUND`

What is wrong?

**Reply by danharrin** [reply, marked as answer]
`Dashboard extends \Filament\Pages\Dashboard`, then remove `$view`
